contributors/devel/api-conventions.md

```diff
@@ -579,21 +579,27 @@ Fields must be either optional or required.
 
 Optional fields have the following properties:
 
-- They have `omitempty` struct tag in Go.
+- They have `+optional` comment tag in Go.
 - They are a pointer type in the Go definition (e.g. `bool *awesomeFlag`) or
 have a built-in `nil` value (e.g. maps and slices).
 - The API server should allow POSTing and PUTing a resource with this field
 unset.
 
+an optional field may also include `omitempty` struct tag in go to omit empty
+values from json serialization. If you want to have different logic for an
+optional field not provided, or provided with empty values, do not use `omitempty`.
+If you don't know what is this, and you just want an optional field, you should
+most probably has `omitempty` too.
```

Comment:
> If you want to have different logic for an optional field not provided, or provided with empty values, do not use omitempty. 

Maybe it's just me, but I find this part a bit hard to digest at a glance. Would you rephrase it?

> If you don't know what is this, and you just want an optional field, you should most probably has `omitempty` too.

Does this mean that in most cases I should add `omitempty` (in addition to `+optional`) to optional fields?